src/mesh/Mesh.cpp

```diff
@@ -29,18 +29,11 @@ Mesh::Mesh(
     : _name(std::move(name)),
       _dimensions(dimensions),
       _id(id),
-      _boundingBox(dimensions)
+      _boundingBox(dimensions),
+      _index(*this)
 {
   PRECICE_ASSERT((_dimensions == 2) || (_dimensions == 3), _dimensions);
   PRECICE_ASSERT(_name != std::string(""));
-
-  meshChanged.connect([](Mesh &m) { query::clearCache(m); });
-  meshDestroyed.connect([](Mesh &m) { query::clearCache(m); });
```

Comment:
When does the clearing happen then? when we call index.clear()?

Reply:
Exactly, this is done in `Mesh::clear()` and when necessary in mappings.